### PR TITLE
fix(gitea): pagination

### LINKED ISSUE #5797: Gitea: wrong pagination handling
https://github.com/renovatebot/renovate/blob/22f76073c3d42d5c5f923a918253eb6f4758492e/lib/platform/gitea/gitea-got-wrapper.ts#L33-L36

when `url.query.page` is `undefined` , `NaN || 1 + 1` is equivalent to `NaN || 2`, so nextPage starts from 3.

~~I have 26 repo, gitea default repo search page size is 10, so repo 21-26 got executed multiple times, and repo 11-20 will never got executed.~~

~~## Fix~~

~~`let nextPage = parseInt(url.query.page || '1', 10)`~~

~~then throw a error when `nextPage` is NaN.~~

seems  modify `url.query` won't affect `URL.format` result

```node
> u = URL.parse('https://git.example.com/api/v1/repos/search?uid=1', true)
Url {
  protocol: 'https:',
  slashes: true,
  auth: null,
  host: 'git.example.com',
  port: null,
  hostname: 'git.example.com',
  hash: null,
  search: '?uid=1',
  query: [Object: null prototype] { uid: '1' },
  pathname: '/api/v1/repos/search',
  path: '/api/v1/repos/search?uid=1',
  href: 'https://git.example.com/api/v1/repos/search?uid=1' }
> u.query.page = '1'
'1'
> URL.format(u)
'https://git.example.com/api/v1/repos/search?uid=1'
```

I have 26 repo, gitea default repo search page size is 10, so repo 1-10 got executed 3 times, and repo 11-26 will never got executed.

### PR DESCRIPTION
**Pull request by NateScarlet**
Closes #5797

### CODE REVIEW

**Review by viceice — CHANGES_REQUESTED** [review]
Please add test case

**Comment by NateScarlet** [comment]
I have difficulties to setup test environment. Can someone write test for me?
image

**Comment by viceice** [comment]
Did you checked: [docs/development/local-development.md](https://github.com/renovatebot/renovate/blob/master/docs/development/local-development.md) ?

**Comment by NateScarlet** [comment]
```powershell
PS D:\~\src\github.com\NateScarlet\renovate> git --version
git version 2.20.1.windows.1
PS D:\~\src\github.com\NateScarlet\renovate> node --version
v12.16.1
PS D:\~\src\github.com\NateScarlet\renovate> yarn --version
1.22.4
PS D:\~\src\github.com\NateScarlet\renovate> python --version
Python 2.7.17
PS D:\~\src\github.com\NateScarlet\renovate> python -c "import mock; print(mock.__version__)"
3.0.5
PS D:\~\src\github.com\NateScarlet\renovate> java -version
java version "1.8.0_201"
Java(TM) SE Runtime Environment (build 1.8.0_201-b09)
Java HotSpot(TM) 64-Bit Server VM (build 25.201-b09, mixed mode)
```

**Comment by viceice** [comment]
can you add the test log, the failed snapshots? whats the difference

**Comment by NateScarlet** [comment]
```
  ● workers/repository/process/lookup › .lookupUpdates() › ignores deprecated

    expect(received).toMatchSnapshot()

    Snapshot name: `workers/repository/process/lookup .lookupUpdates() ignores deprecated 1`

    - Snapshot  - 19
    + Received  +  2

      Object {
        "changelogUrl": undefined,
    -   "deprecationMessage": "On registry `https://registry.npmjs.org/`, the \"latest\" version (v1.4.1) of dependency `q2` has the following deprecation notice:
    -
    - `true`
    -
    - Marking the latest version of an npm package as deprecated results in the entire package being
considered deprecated, so contact the package author you think this is a mistake.",
        "homepage": undefined,
    -   "sourceUrl": "https://github.com/kriskowal/q",
    -   "updates": Array [
    -     Object {
    -       "canBeUnpublished": false,
    -       "fromVersion": "1.3.0",
    -       "isSingleVersion": true,
    -       "newMajor": 1,
    -       "newMinor": 4,
    -       "newValue": "1.4.0",
    -       "releaseTimestamp": "2015-05-09T16:52:40.699Z",
    -       "toVersion": "1.4.0",
    -       "updateType": "minor",
    -     },
    -   ],
    +   "sourceUrl": null,
    +   "updates": Array [],
        "warnings": Array [],
      }

      1060 |         .reply(200, returnJson);
      1061 |       const res = await lookup.lookupUpdates(config);
    > 1062 |       expect(res).toMatchSnapshot();
           |                   ^
      1063 |       expect(res.updates[0].toVersion).toEqual('1.4.0');
      1064 |     });
      1065 |     it('is deprecated', async () => {

      at Object.<anonymous> (lib/workers/repository/process/lookup/index.spec.ts:1062:19)

  ● workers/repository/process/lookup › .lookupUpdates() › is deprecated

    expect(received).toMatchSnapshot()

    Snapshot name: `workers/repository/process/lookup .lookupUpdates() is deprecated 1`

    - Snapshot  - 16
    + Received  +  3

      Object {
        "changelogUrl": undefined,
    -   "homepage": undefined,
    -   "sourceDirectory": "test",
    -   "sourceUrl": "https://github.com/kriskowal/q",
    -   "updates": Array [
    -     Object {
    -       "canBeUnpublished": false,
    -       "fromVersion": "1.3.0",
    -       "isSingleVersion": true,
    -       "newMajor": 1,
    -       "newMinor": 4,
    -       "newValue": "1.4.1",
    -       "releaseTimestamp": "2015-05-17T04:25:07.299Z",
    -       "toVersion": "1.4.1",
    -       "updateType": "minor",
    -     },
    -   ],
    +   "homepage": "http://drarmstr.github.io/c3/",
    +   "sourceUrl": "https://github.com/drarmstr/c3",
    +   "updates": Array [],
        "warnings": Array [],
      }

      1078 |         .reply(200, returnJson);
      1079 |       const res = await lookup.lookupUpdates(config);
    > 1080 |       expect(res).toMatchSnapshot();
           |                   ^
      1081 |       expect(res.updates[0].toVersion).toEqual('1.4.1');
      1082 |     });
      1083 |     it('skips unsupported values', async () => {

      at Object.<anonymous> (lib/workers/repository/process/lookup/index.spec.ts:1080:19)
```
There is too many logs, i just post this two.

**Comment by NateScarlet** [comment]
Maybe these tests fail because i am in china , and i have to use mirror for almost all pacakge manager.

**Comment by viceice** [comment]
the test should work fully offline 🤔

**Comment by NateScarlet** [comment]
I found i can just run jest test for a folder. So i will just let ci do the full test.

**Comment by viceice** [comment]
you can event test a single `.spec.ts`

File: lib/platform/gitea/gitea-got-wrapper.spec.ts [hunk 1]
```diff
@@ -30,25 +30,30 @@ describe('platform/gitea/gitea-got-wrapper', () => {
...
+      });
+    });
 
-    const res = await api.get('pagination-example-1', { paginate: true });
+    const res = await api.get(`${baseURL}/pagination-example-1`, {
+      paginate: true,
+    });
 
     expect(res.body).toHaveLength(6);
     expect(res.body).toEqual(['abc', 'def', 'ghi', 'jkl', 'mno', 'pqr']);
```
Comment (viceice) [inline lib/platform/gitea/gitea-got-wrapper.spec.ts]: you can simplify expect tests above with:

```suggestion
    expect(res.body).toEqual(['abc', 'def', 'ghi', 'jkl', 'mno', 'pqr']);
    expect(got.mock.calls).toMatchSnapshot();
```

**Review by viceice — CHANGES_REQUESTED** [review]
Just some test cleanup / simplification

**Review by viceice — APPROVED** [review]
LGTM

**Pull request merged**